src/AggregateFunctions/DDSketch/Store.h

```diff
@@ -228,9 +228,13 @@ class DDSketchDenseStore : public DDSketchStore
         else
         {
             UInt32 new_length = getNewLength(new_min_key, new_max_key);
+            if (new_length > bins.capacity())
+            {
+                bins.reserve(new_length);
+            }
+            std::size_t old_size = bins.size();
             bins.resize(new_length);
-            bins.resize(bins.capacity());
-            new_max_key = new_min_key + int(bins.size()) - 1;
+            std::fill(bins.begin() + old_size, bins.end(), 0.0);
```

Comment:
Did you mean to put all this back? What's the explanation for the overflow that fuzzer found?

Reply:
I am not sure yet. I will spend sometime troubleshoot the cause.

Reply:
So https://s3.amazonaws.com/clickhouse-test-reports/56342/b55529b5850abc4cc3578b96582c92476c1b558a/fuzzer_astfuzzerubsan/report.html says that this line:
```
int margins = length() - (new_max_key - new_min_key + 1);
```
subtracted 69 from INT32_MIN+68. Presumably the subtraction is `new_max_key - new_min_key`, so new_max_key was INT32_MIN+68, and new_min_key was 69 (:D). So on line 233 int(bins.size()) must've been INT32_MIN, i.e. bins.size() == 2^31.

So I guess it was just very small relative_accuracy and very wide range of values, so max_key - min_key was above 2^30 or so. But that also means that min_key or max_key are probably overflowing all over the place in that test! (Otherwise what are the odds that they came within a factor of 2 of overflowing, but didn't overflow?)

So you need to add overflow checks when calculating keys, throw an exception if it doesn't fit in int32. Or even if it's outside [INT32_MIN/2 + 1, INT32_MAX/2 - 1], so that any two keys can be subtracted without overflow. And do `bins.resize(std::min(bins.capacity(), size_t(INT32_MAX - 1)))`.